src/main/java/Barry.java

```diff
@@ -0,0 +1,322 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Barry {
+
+    // A class for elements in the taskList, with a name and an indicator of whether the task is marked
+    private abstract static class Task{
+        private final String taskName;
+        private boolean isDone;
+
+        // Constructor for the Task class, initialised with the name of the task
+        public Task(String name) {
+            this.taskName = name;
+            this.isDone = false;
+        }
+
+        // Allows the user to modify the status of the task
+        public void modifyStatus(boolean status) {
+            this.isDone = status;
+        }
+
+        // Returns the name of the task
+        protected final String getName() {
+            return this.taskName;
+        }
+
+        // Returns the status of the task
+        protected final boolean getStatus() {
+            return this.isDone;
+        }
+
+        // Returns details about the relevant timings of the task, to be defined in the subclasses
+        abstract public String detailsAsString();
+    }
+
+    // A class that inherits from the Task class, for creating ToDos
+    // (tasks without any date/time attached to them)
+    private static class ToDo extends Task {
+        // Constructor for ToDos, initialised with the name of the task
+        public ToDo(String name) {
+            super(name);
+        }
+
+        // Method which returns details of the object as a String
+        @Override
+        public final String detailsAsString() {
+            return "[T][" + (this.getStatus() ? "X" : " ") + "] " + this.getName();
+        }
+    }
+
+    // A class that inherits from the Task class, for creating Deadlines
+    // (tasks that need to be done before a specific date/time)
+    private static class Deadline extends Task {
+        private final String deadlineDateTime;
+
+        // Constructor for Deadlines, initialised with the name of the task and the deadline
+        public Deadline(String name, String deadline) {
+            super(name);
+            this.deadlineDateTime = deadline;
+        }
+
+        // Method which returns details of the object as a String
+        @Override
+        public final String detailsAsString() {
+            String basicInfo = "[D][" + (this.getStatus() ? "X" : " ") + "] " + this.getName();
+            return basicInfo + " (by: " + deadlineDateTime + ")";
+        }
+    }
```

Comment:
Consider to create a new file called Deadline.java and move the code to the file.